GObject.TypeClass.ref is deprecated in glib 2.84 and removed in glib 2.86. The
[docoumentation][0] instructs to use GObject.TypeClass.get instead.

[0]: https://docs.gtk.org/gobject/type_func.TypeClass.ref.html
diff --git a/lgi/ffi.lua b/lgi/ffi.lua
index 1042632..00d83e5 100644
--- a/lgi/ffi.lua
+++ b/lgi/ffi.lua
@@ -80,7 +80,7 @@ function ffi.load_enum(gtype, name)
    local is_flags = GObject.Type.is_a(gtype, GObject.Type.FLAGS)
    local enum_component = component.create(
       gtype, is_flags and enum.bitflags_mt or enum.enum_mt, name)
-   local type_class = GObject.TypeClass.ref(gtype)
+   local type_class = GObject.TypeClass.get(gtype)
    local enum_class = core.record.cast(
       type_class, is_flags and GObject.FlagsClass or GObject.EnumClass)
    for i = 0, enum_class.n_values - 1 do
